src/editors/containers/ProblemEditor/components/EditProblemView/index.jsx

```diff
@@ -35,30 +35,32 @@ export const EditProblemView = ({
   // injected
   intl,
 }) => {
+  const dispatch = useDispatch();
   const editorRef = useRef(null);
   const isAdvancedProblemType = problemType === ProblemTypeKeys.ADVANCED;
-  const { isNoAnswerModalOpen, openNoAnswerModal, closeNoAnswerModal } = noAnswerModalToggle();
-  const dispatch = useDispatch();
+  const { isSaveWarningModalOpen, openSaveWarningModal, closeSaveWarningModal } = saveWarningModalToggle();
 
   return (
     <EditorContainer
       getContent={() => getContent({
         problemState,
-        openNoAnswerModal,
+        openSaveWarningModal,
         isAdvancedProblemType,
         editorRef,
         assets,
         lmsEndpointUrl,
       })}
     >
       <AlertModal
-        title={intl.formatMessage(messages.noAnswerModalTitle)}
-        isOpen={isNoAnswerModalOpen}
-        onClose={closeNoAnswerModal}
+        title={isAdvancedProblemType ? (
```

Comment:
Correct, the no answer modal only opens in the simple editor because we don't parse the content of the raw OLX. Therefore, we don't know if there are answers present or not.